### PR TITLE
(#58) Support Authorization header as well as X-ZUMO-AUTH

### PR DESCRIPTION
**Pull request by adrianhall**

### CODE REVIEW

File: src/express/middleware/authenticate.js [hunk 1]
```diff
@@ -29,31 +29,39 @@ module.exports = function (configuration) {}
...
-            var token = req.get('x-zumo-auth');
+        return function(req, res, next) {
+            // If the X-ZUMO-AUTH header is there, use that.  If the X-ZUMO-AUTH
+            // header is not there, then look for a Bearer Authorization header.
+            var token = req.get('x-zumo-auth') || req.get('authorization');
+
+            if (token) {
+                // Check to see if the token has the bearer in the front - Authorization
+                // will, X-ZUMO-AUTH won't.  Remove it if present
+                if (token.toLowerCase().substr(0, 7) === 'bearer ') {
```
Comment (shrishrirang) [inline src/express/middleware/authenticate.js]: Shouldn't we trim? Isn't "        bearer xyz" (space as prefix) a valid value?

Comment (adrianhall) [inline src/express/middleware/authenticate.js]: We are just comparing the first 7 letters to the string provided.  It has to be bearer with a space after it.  All the bearer character could conceivably be in a base64 token, whereas space isn't.  Admittedly, I'd be shocked if a base64 token started with those characters, but without the space test, we have a wierd corner case.

Comment (danderson00) [inline src/express/middleware/authenticate.js]: I can't see in the spec where it allows for spaces. The header must be prefixed by "bearer " for digest authentication. I think this implementation is fine.

**Comment by danderson00** [comment]
It looks good, though maybe we should add an integration test.

**Comment by danderson00** [comment]
Far more comprehensive than expected! Awesome - :shipit:
